code/src/Components/FilmList.jsx

```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import styled from 'styled-components'
+
+const FilmList = ({films}) => {
+    return(
+        <Main>
+        {films.map((film) => {
+        return(
+        <Link to={`/movies/${film.id}`} key={film.id}>
+        <Filmcard>
+            <FilmImageGroup>
+                <CardImage className="card-image" src={`https://image.tmdb.org/t/p/w500${film.poster_path}`} alt={film.title} />
+                <FilmImageOverlay></FilmImageOverlay>
+                <FilmTextOverlay>
+                    <FilmTitle>{film.title}</FilmTitle>
+                    <h3>Released {film.release_date}</h3>
+                </FilmTextOverlay>
+            </FilmImageGroup>
+        </Filmcard>
+        </Link>
+        )
+        })}
+        </Main>
+    )
+}
+
+export default FilmList
+
+// Styled Components
+const Main = styled.main`
+  display: grid;
+  grid-template-columns: 1fr;
+
+  @media (min-width: 768px) {
+    grid-template-columns: 1fr 1fr;
+  }
+
+  @media (min-width: 1100px) {
+    grid-template-columns: 1fr 1fr 1fr 1fr;
+  }
+`
+
+const Filmcard = styled.div`
+  width: 100%;
+  height: 100%;
+`
+
+// NEEDS HOVER
+const FilmImageOverlay = styled.div`
+  position: absolute;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  transition: 0.5s ease;
+  background: transparent;
+`
+
+const FilmTextOverlay = styled.div`
+  position: absolute;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  transition: 0.5s ease;
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-end;
+  color: transparent;
+  padding: 20px;
+`
+
+const FilmImageGroup = styled.div`
+  position: relative;
+  display: flex;
+  width: 100%;
+  height: 100%;
+
+  &:hover ${FilmImageOverlay} {
```

Comment:
Interesting way of adding style with javascript. Curious to know if you had to google a long time to find this solution.